homework/index.js

```diff
@@ -30,15 +32,102 @@
     return elem;
   }
 
+  function renderError(error) {
+    const root = document.getElementById('root');
+    createAndAppend('h1', root, { text: error.message });
+  }
+
+  function createOptions(repositoryDetails, select) {
+    repositoryDetails.forEach((repository, index) => {
+      createAndAppend('option', select, {
+        text: repository.name,
+        value: index,
+      });
+    });
+  }
+
+  function addRow(tbody, labelText, value, str) {
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, { text: `${labelText}`, class: str });
+    createAndAppend('td', tr, { text: value });
+    return tr;
+  }
+
+  function renderRepoInformation(repositories, container) {
+    container.innerHTML = ' ';
+    const table = createAndAppend('table', container, { class: 'infoTable' });
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, 'Repository: ', '', 'explanations');
+    createAndAppend('a', firstRow.lastChild, {
+      href: `${repositories.html_url}`,
+      target: 'blank',
+      text: `${repositories.name}`,
+    });
+    addRow(tbody, 'Description:', repositories.description, 'explanations');
+    addRow(tbody, 'Fork:', repositories.forks, 'explanations');
+    addRow(tbody, 'Updated:', `${repositories.updated_at}`.substring(0, 10), 'explanations');
+  }
+
+  function renderContributorsInformation(repositories, container) {
+    const contributors = repositories.contributors_url;
+    const url = contributors;
+    fetchJSON(url)
+      .then(contributorDetails => {
+        console.log(contributorDetails);
+        container.innerHTML = ' ';
+        const div = document.getElementById('cont-ulist');
+        createAndAppend('h4', div, { text: 'Contributors' });
+        const contrList = createAndAppend('ul', div, { class: 'contr-list' });
+        contributorDetails.forEach(contributor => {
+          const li = createAndAppend('li', contrList);
+          const a = createAndAppend('a', li, { href: contributor.html_url, target: 'blank' });
```

Comment:
I was surprised that `blank` worked here. But better use `_blank` to be safe.

https://stackoverflow.com/questions/35703005/what-is-the-difference-between-target-blank-and-target-blank